main.py

```diff
@@ -0,0 +1,220 @@
+            date: {
+                fields:  value
+            }    
+        }
+    ]
+"""
+class ReadFilesData:
+    def __init__(self, year, month):
+        self.year = year
+        self.month = month
+        self.data = []
+        self.readDatadromFiles()
+        
+    """_summary_
+        read data from every single file 
+        if month is None then it iterate over all the months that is present in the months global variable
+        if month is present then it only iterate over the single month
+    """
+    def readDatadromFiles(self):
+        if self.month == None:
+            for month in months.values():
+                self.readDataFromSingleFile(month)
+                
+        else: 
+            self.readDataFromSingleFile(months[self.month])
+
+        if len(self.data) == 0:
+            raise Exception("No file is found")
+            
+    
+    """
+    This function is make connection with the single file and then read the data line by line from the file
+    it read first line and set the column names or the fields name from the first line of the file and then
+    set it in the col_name to make dict
+    then it read the other lines one by one and then pick the field name from col_name and value from the 
+    temperature record (temp_record) and make the dict
+    
+    """    
+    def readDataFromSingleFile(self, month):
+        try :
+            with open('weather_reports/Murree_weather_{}_{}.txt'.format(self.year, month), 'r') as file:
+                single_file_records = {}
+                single_line = file.readline()
+                _, *column_names = single_line.split(',')
+                single_date_record = {}
+                
+                while True:
+                    line = file.readline()
+                    if not line:
+                        break
+                    else:
+                        date, *temperature_records = line.split(',')
+                        for index in range(len(temperature_records)):
+                            single_date_record[column_names[index]] = temperature_records[index]
+                            
+                        single_file_records.update({date: single_date_record})
+                        single_date_record = {}  
+                self.data.append(single_file_records)
+        except : 
+            print('{}/{} file is not exist'.format(month, self.year))
+            
+        
+class TemperatureCalculation:
+    def __init__(self, data_from_files):
+        self.data = data_from_files
+        
+    def requiredTemperatureCalculation(self, type_of_calculation):
+        """
+        call function according to the input
+        """
+        temperature_field = TEMPERATURE_FIELDS[type_of_calculation]
+        return self._temperatureCalculationFromFiles(temperature_field["name"], temperature_field["maximum_number"])
+    
+    def _temperatureCalculationFromFiles(self, field, maxNumber):
```

Comment:
Lets rename these, or are they reading from files?

Reply:
it is renamed